h2o-algos/src/main/java/hex/tree/DHistogram.java

```diff
@@ -331,6 +341,8 @@ else if (_histoType== SharedTreeModel.SharedTreeParameters.HistogramType.Quantil
         _splitPtsFloat[i] = (float) binAt(i);
       }
     }
+    assert !_intOpt || _splitPts == null : "Integer-optimization cannot be enabled when split points are defined";
```

Comment:
You check  `_splitPts == null` here in the assert, but I can't see this check in `useIntOpt()` method. There should be something like (similar to constraints):

```
    if (_splitPts != null) {
      return false;
    }
```

Or am I missing something?

Reply:
I got it, the _splitPts is initialized later (before this assert).